test/system/react_components/notifications/icon_test.rb

```diff
@@ -0,0 +1,23 @@
+require "application_system_test_case"
+
+module Notifications
+  class IconTest < ApplicationSystemTestCase
+    test "increments number when a notification is created" do
+      create(:user)
+
+      visit test_components_notifications_icon_path
+      click_on "Create notification"
+
+      assert_text "Count\n1", wait: 1
+    end
+
+    test "shows unread state when a new notification is created" do
+      create(:user)
+
+      visit test_components_notifications_icon_path
+      click_on "Create notification"
+
+      assert_text "Unread\ntrue", wait: 1
```

Comment:
`wait: 1` is nice. Could you refactor the other websockets tests to have that too pls?

Reply:
(can be a seperate PR)

Reply:
I tried this and it's a bit more complicated. The fact that we click a button here allows the websockets to connect just in time. The other tests don't have that and would still need the sleep at the start of the test to establish the websocket connection. The second sleep that we have in those tests could also be deleted since assertions wait for a certain amount of time by default.

I will keep the existing tests as they are for now and will refactor later.